### PR TITLE
Reconstruct `test_psnr.py`

### PR DESCRIPTION
**Pull request by puhuk**
Description: Reconstruct `test_psnr.py`

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/metrics/test_psnr.py [hunk 1]
```diff
@@ -79,144 +79,344 @@ def test_psnr():
...
+        y_pred,
+        y,
+        data_range,
+        metric_device,
+        n_iters,
+        batch_size,
+        atol,
+        output_transform=lambda x: x,
+        compute_y_channel=False,
+    ):
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_psnr.py:93]: Let's try to eliminate duplicated code

Comment (puhuk) [reply tests/ignite/metrics/test_psnr.py:93]: Do you mean to extract the `def _test` outside the test code and be used commonly from other test methods?

Comment (vfdev-5) [reply tests/ignite/metrics/test_psnr.py:93]: Any solution work, just do not add duplicated code.

File: tests/ignite/metrics/test_psnr.py [hunk 2]
```diff
@@ -123,100 +125,160 @@ def update(engine, i):
...
 
-    # test for float
-    manual_seed(42)
-    y_pred = torch.rand(offset * idist.get_world_size(), 3, 28, 28, device=device)
-    y = y_pred * 0.65
-    data_range = (y.max() - y.min()).cpu().item()
-    _test(y_pred, y, data_range, "cpu", n_iters, s, offset, rank, atol=atol)
+    for i in range(3):
+        # check multiple random inputs as random exact occurencies are rare
+        rank = idist.get_rank()
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_psnr.py:141]: let's define rank outside the loop

File: tests/ignite/metrics/test_psnr.py [hunk 3]
```diff
@@ -123,100 +125,160 @@ def update(engine, i):
...
+        torch.manual_seed(42 + rank + i)
+        y_pred, y = get_test_cases()
+        _test(
+            y_pred,
+            y,
+            1,
+            "cpu",
+            n_iters,
+            batch_size,
+            atol=atol,
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_psnr.py:151]:
```suggestion
            atol=atol
```

Comment (puhuk) [reply tests/ignite/metrics/test_psnr.py:151]: When I eliminate the last comma, but when I run `run_code_style.sh` it reverts.
Let me manually change the line and send again.

Comment (vfdev-5) [reply tests/ignite/metrics/test_psnr.py:151]: No, it works. For complicated case with lambda function you can do something like
```python
    def out_fn(x):
        return x[0][:, 0, ...], x[1][:, 0, ...]

    rank = idist.get_rank()
    for i in range(3):
        # check multiple random inputs as random exact occurencies are rare
        torch.manual_seed(42 + rank + i)
        y_pred, y = get_test_cases()
        _test(y_pred, y, 220, "cpu", n_iters, batch_size, atol, output_transform=out_fn, compute_y_channel=True)
        if device.type != "xla":
            dev = idist.device()
            _test(y_pred, y, 220, dev, n_iters, batch_size, atol, output_transform=out_fn, compute_y_channel=True)

```

File: tests/ignite/metrics/test_psnr.py [hunk 4]
```diff
@@ -123,100 +125,105 @@ def update(engine, i):
...
+        return x[0][:, 0, ...], x[1][:, 0, ...]
+
+    rank = idist.get_rank()
+    for i in range(3):
+        # check multiple random inputs as random exact occurencies are rare
+        torch.manual_seed(42 + rank + i)
+        y_pred, y = get_test_cases()
+        _test(y_pred, y, 220, "cpu", n_iters, batch_size, atol, output_transform=out_fn, compute_y_channel=True)
+        if device.type != "xla":
+            _test(
```
Comment (puhuk) [inline tests/ignite/metrics/test_psnr.py:174]: This line can not be written with 1 line, is there any method to deal with?

Comment (vfdev-5) [reply tests/ignite/metrics/test_psnr.py:174]: Check this comment https://github.com/pytorch/ignite/pull/2679#discussion_r961583454 
It does not work ?

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @puhuk !

**Pull request merged**